src/plugins/maps-dashboards-plugin/public/map/service_settings.js

```diff
@@ -0,0 +1,328 @@
+      fetchFunction: function (...args) {
+        return fetch(...args);
+      },
+    });
+    this.getTMSOptions();
+  }
+
+  getTMSOptions() {
+    const markdownIt = new MarkdownIt({
+      html: false,
+      linkify: true,
+    });
+
+    // TMS Options
+    this.tmsOptionsFromConfig = _.assign({}, this._tilemapsConfig.options, {
+      attribution: _.escape(markdownIt.render(this._tilemapsConfig.options.attribution || '')),
+      url: this._tilemapsConfig.url,
+    });
+  }
+
+  shouldShowZoomMessage({ origin }) {
+    return origin === ORIGIN.EMS && this._showZoomMessage;
+  }
+
+  enableZoomMessage() {
+    this._showZoomMessage = true;
+  }
+
+  disableZoomMessage() {
+    this._showZoomMessage = false;
+  }
+
+  __debugStubManifestCalls(manifestRetrieval) {
+    this._emsClient = this._opensearchMapsClient;
+    const oldGetManifest = this._emsClient.getManifest;
+    this._emsClient.getManifest = manifestRetrieval;
+    return {
+      removeStub: () => {
+        delete this._emsClient.getManifest;
+        //not strictly necessary since this is prototype method
+        if (this._emsClient.getManifest !== oldGetManifest) {
+          this._emsClient.getManifest = oldGetManifest;
+        }
+      },
+    };
+  }
+
+  _backfillSettings = (fileLayer) => {
+    // Older version of OpenSearch Dashboards stored EMS state in the URL-params
+    // Creates object literal with required parameters as key-value pairs
+    const format = fileLayer.getDefaultFormatType();
+    const meta = fileLayer.getDefaultFormatMeta();
+
+    return {
+      name: fileLayer.getDisplayName(),
+      origin: fileLayer.getOrigin(),
+      id: fileLayer.getId(),
+      created_at: fileLayer.getCreatedAt(),
+      attribution: getAttributionString(fileLayer),
+      fields: fileLayer.getFieldsInLanguage(),
+      format: format, //legacy: format and meta are split up
+      meta: meta, //legacy, format and meta are split up
+    };
+  };
+
+  // anyone using this._emsClient should call this method before, to set the right client
+  async _setMapServices() {
+    // if client is not null, return immediately.
+    // Effectively, client creation will be called only once.
+    if (this._emsClient) {
+      return;
+    }
+    const useOpenSearchMaps = await this._opensearchMapsClient.isEnabled();
+    if (useOpenSearchMaps) {
+      // using OpenSearch Maps.
+      this._emsClient = this._opensearchMapsClient;
+    } else {
+      // not using OpenSearch Maps, fallback to default maps.
+      this._emsClient = new EMSClient({
+        language: i18n.getLocale(),
+        appVersion: getOpenSearchDashboardsVersion(),
+        appName: 'opensearch-dashboards',
+        fileApiUrl: this._mapConfig.emsFileApiUrl,
+        tileApiUrl: this._mapConfig.emsTileApiUrl,
+        landingPageUrl: this._mapConfig.emsLandingPageUrl,
+        fetchFunction: function (...args) {
+          return fetch(...args);
+        },
+      });
+    }
+  }
+
+  async getFileLayers() {
```

Comment:
same here, we will need to use the latest bug fix code. https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1550/files#diff-edd35a21ee8bbae0d475be1ab224deda046e45c13d0d09b752c9b43b7da6a2ff